### PR TITLE
[Intl] Allow compressing emoji and data maps

### LINKED ISSUE #49943: Split intl packages in smaller packages
### Description

Right now Symfony Intl package is 43MB, which makes all Applications really big 
image

The problems for me are:

- Packaging for Bref can be a problem as AWS has a limit of 250 MB (unzipped)
- Waste of Traffic and Disk Space

### Example

My ideas so far:

- Create separate packages for the actual data emoji, locales, timezones and optin people for each feature (symfony/intl-data-emoji)
- Or complete symfony/intl-emoji like split the repository on Intl level again

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Fix #49943
| License       | MIT
| Doc PR        | -

Resuming #49970, which was closed by mistake :)

### CODE REVIEW

**Comment by stof** [comment]
My review at https://github.com/symfony/symfony/pull/49970#pullrequestreview-1390340803 applies to this PR then.

File: src/Symfony/Component/Intl/Resources/bin/compress [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env php
+<?php
+
```
Comment (dkarlovi) [inline src/Symfony/Component/Intl/Resources/bin/compress:3]: Copyright header missing.

File: src/Symfony/Component/Intl/Resources/emoji/build.php [hunk 2]
```diff
@@ -178,7 +178,7 @@ public static function buildStripRules(array $emojisCodePoints): iterable
             $maps[$codePointsCount][$emoji] = '';
         }
 
-        return ['strip' => self::createRules($maps)];
+        return ['emoji-strip' => self::createRules($maps)];
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Intl/Resources/emoji/build.php:181]: those changes are not directly related, although I found them while running the script

**Review by dkarlovi — APPROVED** [review]

File: src/Symfony/Component/Intl/Resources/bin/compress (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Intl/Resources/bin/compress:3]: not anymore :)

**Comment by nicolas-grekas** [comment]
CI updated to run the intl tests with compressed data.

File: .github/workflows/unit-tests.yml [hunk 3]
```diff
@@ -221,6 +221,15 @@ jobs:
         run: |
             script -e -c './phpunit --group tty' /dev/null
 
+      - name: Intl tests with compressed data
```
Comment (stof) [inline .github/workflows/unit-tests.yml]: Should this be in this workflow or in the special one about icu data ? Many intl tests are excluded in this workflow.

Comment (nicolas-grekas) [reply .github/workflows/unit-tests.yml]: good idea, updated
